keps/sig-release/1731-publishing-packages/README.md

```diff
@@ -151,49 +163,185 @@ Scenario: [...]
+Packages will be built and published using [Open Build Service (OBS)][obs]. openSUSE will sponsor the Kubernetes
+project by giving us access to the [OBS instance hosted by openSUSE][obs-build].
+
+[obs]: https://openbuildservice.org/
+[obs-build]: https://build.opensuse.org/
+
+#### Using OBS instead of manually building and hosting packages
+
+The reasons for using Open Build Service (OBS) instead of building and hosting packages ourselves are:
+
+- We want to handoff managing GPG keys to the third-party
+  - Managing GPG keys ourselves represents a security risk. For example, if a Release Manager with access to the GPG key
+    steps down, we might need to rotate the key. This is a process that affects End Users, therefore
+    we want to avoid it
+  - In this case, GPG keys are securely managed by the OBS platform hosted by openSUSE. No one from the Kubernetes
+    project will have direct access to the key, mitigating one of the main risks of this proposal
+- We want to avoid managing the infrastructure ourselves, including buckets, mirrors/CDNs...
+- We want to provide 'as a service' access to the packages infrastructure to Release Managers and eventually other
+  Kubernetes maintainers for their subprojects
+
+#### How Open Build Service works?
+
+From the [OBS website](https://openbuildservice.org/):
+
+> The Open Build Service (OBS) is a generic system to build and distribute binary packages from sources in an automatic, consistent and reproducible way. You can release packages as well as updates, add-ons, appliances and entire distributions for a wide range of operating systems and hardware architectures.
+
+OBS works in a way that we push sources and package spec files. Upon pushing packages/changes, OBS automatically
+triggers builds for all chosen operating systems and architectures. Under the hood, OBS uses the same set of tools that
+we use for building packages: `dpkg-buildpackage` and `rpmbuild`.
+
+OBS implements a simple source-control management (SCM) system. It provides a complete history for all packages
+allowing users to see what spec files and sources we used to build the concrete package. The history is accessible
+via the OBS web interface.
+
+Interaction with the OBS platform is done mainly via the [`osc` command-line tool][osc]. Alternatively, it's possible
+to interact via the web interface. Currently there are no (Go) libraries that we can use instead of the `osc` tool.
+
+[osc]: https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.osc.html
+
+#### OBS configuration and packages layout
+
+We'll publish Debian-based (`deb`) and RPM-based (`rpm`) packages. Packages will be published for the following
+architectures: `aarch64`, `armv7l`, `ppc64le`, `s390x`, `x86_64`. We'll build packages on the following operating
+systems and architectures:
+
+- Ubuntu 20.04 (`aarch64`, `armv7l`, `ppc64le`, `s390x`, `x86_64`)
+- CentOS Stream 8 (`aarch64`, `ppc64le`, `x86_64`)
+- OpenSUSE Factory (`armv7l`, `s390x`)
```

Comment:
Ongoing Slack discussion: https://kubernetes.slack.com/archives/C03U7N0VCGK/p1674553212858809

Reply:
there is sles packages for s390x in the OBS

Reply:
Agreed, it might be better to pick something older  like SLE15 or openSUSE Step 15. does this have to be explicitly listed here?

Reply:
Yes. So probably the openSUSE Leap 15.0 would work for this purpose.

Reply:
15.4/15SP4

Reply:
Since SUSE 15 minor versions don't retain full userspace ABI compatibility, it only makes sense to target the absolute oldest so that we have maximum compatibility.

Reply:
As far as I see, `openSUSE Leap 15.3` is the oldest available on OBS. Should we target that one or do we need earlier/older?

Reply:
Looks like openSUSE Leap 15.3 was the oldest with s390x, but 15.3 and 15.4 are ABI compatible, so we can just use 15.4.

```
  <repository name="openSUSE_Leap_15.4">
    <path project="openSUSE:Leap:15.4" repository="standard"/>
    <arch>s390x</arch>
  </repository>

```

Reply:
Sle15:ga should be available for s390x as well

Reply:
What’s the difference between GA and 15.3?

Reply:
I don't think there is a significant difference in available macros between either 15.0 and 15.4, and if there is, it shouldn't decide on this imho. we can always add macros just for `isv:kubernetes`.

If we have a concern about interoperability, then we need to pick a really old but still maintained distribution. For example SLE12 SP5 exists in OBS as well. I'd pick that over SLE15.0 (which is now EOL). 15.4+ should be fine, but that's why I was asking why we hardcode the versions here. Eventually only 15.5 will be supported and we should switch to that then. 

So either we want to hardcode to something specific, then pick SLE12 SP5. or you want something reasonable and maintained, then chose SLE15 (refering to the at any point in time latest SLE15X). Essentially with "Centos Stream 8" you're doing exactly the same. it isn't referring to a particular release but to the tip of a rolling release distribution.

Reply:
If we rely on OBS specific features, then it becomes harder for downstreams to rebuild our packaging in their own infrastructure. While it's true that it's possible to deploy an OBS, not everyone wants to do that for _just_ Kubernetes. So in-distro packaging, macros, and RPM features matter here.

Reply:
First of all, thank you so much folks for such valuable feedback! I'll implement two changes regarding this:

- We'll try SLE 12 SP5 instead of openSUSE Factory to provide better compatibility
- We'll clearly state that builders can change at any time without prior notification and that it should be only used as a guidance for the initial implementation